frontend/src/metabase/api/database-candidates.ts

```diff
@@ -0,0 +1,16 @@
+import { Api } from "metabase/api";
+import type { DatabaseCandidate, DatabaseId } from "metabase-types/api";
+
+// a "database candidate" is a set of sample x-rays we suggest for new users
+export const databaseCandidatesApi = Api.injectEndpoints({
+  endpoints: builder => ({
+    listDatabaseCandidates: builder.query<
+      DatabaseCandidate[],
+      { id?: DatabaseId }
```

Comment:
We should make it required and use [`skipToken`](https://redux-toolkit.js.org/rtk-query/usage-with-typescript#skipping-queries-with-typescript-using-skiptoken).

```suggestion
      { id: DatabaseId }
```

----

If we only have a single parameter, we can skip the object:

```suggestion
      DatabaseId
```